### PR TITLE
[Feature] Skill Picker Dialogs

### LINKED ISSUE #5053: Skill Picker v3
## 👋 Introduction

Create the new version of the skill picker based on specifications/prototypes to be provided by @substrae 

> **Warning**
> We need these ready for the testing being done 19-12-22

## ✅ Acceptance Criteria

- [ ] Three separate skill picker dialogs (stories only)
    - [ ] Find a skill (search form, pools)
    - [ ] Add skill to experience (experience forms)
    - [ ] Add skill to library (new page yet to be added)

### PR DESCRIPTION
**Pull request by esizer**
🤖  Resolves #5053 

## 👋 Introduction

Adds a new `SkillDialog` component.

## 🕵️ Details

The new component is a form that, when submitted, returns a skill, level and if it is a current skill.

### API

```tsx
type SkillDialogContext = "experience" | "library";

export interface SkillDialogProps {
  skills: Skill[]; // Available Skills
  context?: SkillDialogContext; //The context in which this dialog appears
  showCategory?: boolean; // If we should allow users to filter skills and families by category
  trigger?: {
    label?: React.ReactNode;
    icon?: IconType;
  }; // Allows us to display a custom trigger
  onSave: (values: FormValues) => Promise<void>; // Callback when user submits the form
}
```

## ✅ TODO

- [x] Send copy off for translation
- [x] Add translations

## 🧪 Testing

>**Note**
> We currently do not have the definitions for the skill levels. Once they have been provided, we can add them

1. Run storybook `npm run storybook`
2. Navigate to the "Components > Dialog > Skill Dialog" section
3. Interact with the dialogs
4. Confirm they follow the [spec outlined in the issue ](https://github.com/GCTC-NTGC/gc-digital-talent/issues/5053)

### Default
Screenshot 2023-06-16 093704

## Experience
Screenshot 2023-06-16 093805

## Library
Screenshot 2023-06-16 093839

### CODE REVIEW

File: apps/web/src/components/SkillDialog/SkillSelection.tsx [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+        {...(showCategory
+          ? {
+              "data-h2-grid-template-columns": "base(1fr) p-tablet(1fr 1fr)",
+            }
+          : {
+              "data-h2-grid-template-columns": "base(1fr)",
+            })}
+      >
+        {showCategory && (
+          <Select
```
Comment (petertgiles) [inline apps/web/src/components/SkillDialog/SkillSelection.tsx:99]: It seems like the _category_ selection has no effect if a skill family is selected.  They are not "anding" or "oring".  Maybe the _category_ filter could be reset to blank when a skill family is selected to show what is happenign?

**Review by petertgiles — APPROVED** [review]
Looks great!  I had a few suggestions but nothing blocking.

File: apps/web/src/components/SkillDialog/SkillSelection.tsx (hunk 1 above)
Comment (esizer) [reply apps/web/src/components/SkillDialog/SkillSelection.tsx:99]: Oh, this is a good catch. I think I completely messed this up 😓 @substrae are you able to confirm what is intended to happen when a user changes either filter?

File: apps/web/src/components/SkillDialog/SkillSelection.tsx [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+      </div>
+      <div data-h2-margin="base(x1, 0)">
+        <Combobox
+          id="skill"
+          name="skill"
+          rules={{ required: intl.formatMessage(errorMessages.required) }}
+          trackUnsaved={false}
+          label={intl.formatMessage(
+            {
+              defaultMessage: "Select from {count} matching skills",
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/components/SkillDialog/SkillSelection.tsx]: Updated to be pluralized 👍 

[pluralize skill label](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5054/commits/13193a0ff1775d8c80b6913d2629215235bafb09)

File: apps/web/src/components/SkillDialog/SkillDialog.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+export default {
+  component: SkillDialog,
+  title: "Components/Dialog/Skill Dialog",
+  decorators: [OverlayOrDialogDecorator],
+  args: {
+    skills: mockSkills,
+  },
+};
+
+const Template: StoryFn<typeof SkillDialog> = (args) => {
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/components/SkillDialog/SkillDialog.stories.tsx:22]: Excellent idea! 💡 Updated in [default to open for storybook](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5054/commits/67edad239f0f6636f4077d275605ea408a2e63c2)

File: apps/web/src/components/SkillDialog/SkillSelection.tsx (hunk 1 above)
Comment (substrae) [reply apps/web/src/components/SkillDialog/SkillSelection.tsx:99]: @esizer @petertgiles 

To be honest I'm not sure how skill families and categories interact with each other, but the way I expected this to work was that filtering by category would only show the skill families in that category, and only the skills in that category. Selecting a skill family and all categories and then choosing a narrower category would reset skill families in the event that the category didn't contain the existing skill family.

Comment (petertgiles) [reply apps/web/src/components/SkillDialog/SkillSelection.tsx:108]: Oh yeah, this is nice.  A+.

**Comment by substrae** [comment]
@esizer 

I can't remember if I mentioned this to you, but ideally when you focus the skill search input, the dropdown would open immediately. That way, users have options to select from right off the bat, without the need to start typing or manually select the chevron button.

image

One last thing - any chance we can increase the gap distance between these two fellas? Ideally it would mirror the distance of the solid button's right-hand padding.

image

**Comment by substrae** [comment]
@esizer

Shucks, okay. Let's keep an eye on that discussion. To be frank, this is almost enough for me to say we should use a different third party component. The UX is extremely bad without it.

**Pull request merged**